shared/domains/chelonia/encryptedData.js

```diff
@@ -277,6 +277,51 @@ export const encryptedIncomingForeignData = <T>(contractID: string, _0: any, dat
   })
 }
 
+export const encryptedIncomingDataWithRawKey = <T>(key: Key, data: any, additionalData?: string): EncryptedData<T> => {
+  if (data === undefined || !key) throw new TypeError('Invalid invocation')
+
+  let decryptedValue
+  const eKeyId = keyId(key)
+  const decryptedValueFn = (): any => {
+    if (decryptedValue) {
+      return decryptedValue
+    }
+    const state = {
+      _vm: {
+        authorizedKeys: {
+          [eKeyId]: {
+            purpose: ['enc'],
+            data: serializeKey(key, false),
+            _notBeforeHeight: 0,
+            _notAfterHeight: undefined
+          }
+        }
+      }
+    }
+    decryptedValue = decryptData.call(state, NaN, data, { [eKeyId]: key }, additionalData || '')
+
+    return decryptedValue
+  }
+
+  return wrapper({
+    get encryptionKeyId () {
+      return encryptedDataKeyId(data)
+    },
+    get serialize () {
+      return () => data
+    },
+    get toString () {
+      return () => JSON.stringify(this.serialize())
+    },
```

Comment:
I realize I didn't ask this when you sent in the original PR that introduced all of these concepts (or at least, I don't remember asking), but why is it that `serialize` returns the input encrypted `data` instead of the decrypted version of it? Maybe that can be explained in a comment somewhere?

Reply:
Comments added at the top of the file. `serialize` returns encrypted data because `encryptedData` is meant as a container for data. It just seems logical to me. Also, this is called from `GIMessage.serialize()`. To get the plaintext, you'd have to call `valueOf()`